blockchain/blockchain_data_structure.py

```diff
@@ -83,28 +110,35 @@ def get_latest_block(self):
 
     def mine_pending_transactions(self):
         latest_block_index = self.get_latest_block().index + 1
-        block = Block(datetime.now().strftime("%m/%d/%Y, %H:%M:%S"), self.pending_transactions,
+        block_trans = self.pending_transactions.copy()
+        block = Block(datetime.now().strftime("%m/%d/%Y, %H:%M:%S"), block_trans,
```

Comment:
I believe the timestamp, datetime.now() might be a better internal representation for the blockchain; if needed, we can convert it to another time format on the frontend